components/api-server/src/bin/api_server.rs

```diff
@@ -0,0 +1,100 @@
+use axum::{
+    Json,
+    Router,
+    extract::{Path, State},
+    http::StatusCode,
+    response::{
+        IntoResponse,
+        Sse,
+        sse::{Event, KeepAlive},
+    },
+    routing::{get, post},
+};
+use clap::Parser;
+use clp_rust_utils::package_config;
+use futures::{Stream, StreamExt};
+use thiserror::Error;
+use tracing_subscriber::{self, prelude::*};
+
+#[derive(Parser)]
+#[command(version, about)]
+struct Args {
+    #[arg(long)]
+    package_root: String,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::registry()
+        .with(tracing_subscriber::fmt::layer())
+        .with(tracing_subscriber::EnvFilter::from_default_env())
+        .init();
+
+    let args = Args::parse();
+    let package_root = std::path::Path::new(&args.package_root);
+
+    let config_path = package_root.join(package_config::DEFAULT_CONFIG_FILE_PATH);
+    let config_string = std::fs::read_to_string(config_path)?;
+    let config: package_config::Config = serde_yaml::from_str(&config_string)?;
+
+    let credentials_path = package_root.join(package_config::DEFAULT_CREDENTIALS_FILE_PATH);
+    let credentials_string = std::fs::read_to_string(credentials_path)?;
+    let credentials: package_config::Credentials = serde_yaml::from_str(&credentials_string)?;
```

Comment:
would be be better to create helpers from those and maintain them in `clp-rust-utils`, so they become reusable for any project / binary that needs to parse clp config / credentials from files?